src/transformers/modeling_bert.py

```diff
@@ -409,35 +409,19 @@ def forward(
     ):
         all_hidden_states = ()
         all_attentions = ()
+        output_attentions = torch.tensor(output_attentions, dtype=torch.bool)
```

Comment:
Is it correct to have this tensor on the cpu or is `torch.utils.checkpoint.checkpoint` expecting everything to be on the same device? Also, it might be worth trying it with ddp just to be sure. 

Another thing to check is pytorch-xla. It probably doesn't support gradient checkpointing in the first place, but just checking that a mix of cpu/tpu tensors is not an issue.

Reply:
Alternatively, is it possible to generalize the following wrapper of the forward function to automatically handle non-tensor arguments? 
```
def create_custom_forward(module):
    def custom_forward(*inputs):
        return module(*inputs, output_attentions)
    return custom_forward
```